### ISSUE TITLE
Sporadisch keine Aktualisierung von Objektdaten

### ISSUE THREAD

**Issue by joergpichler** [stoffel7/ioBroker.enet#32]
Ich habe mit der aktuellen Version das Problem, dass wenn ich im eNet Server eine Szene triggere z.B. alle Jalousien im Wohnzimmer hoch/runter das ein einzelnes Objekt die Änderungen nicht mitbekommt. Z.B. eine Jalousie ist physisch komplett geöffnet, das Objekt bleibt aber auf 60% hängen. Die eNet App zeigt die korrekten Werte an.

Was für Daten kann ich potentiell zum Debuggen beitragen?

**Comment by stoffel7** [comment]
was wird denn im log ausgegeben, wenn die Jalousien sich schließen? ( es müßten mehrere Werte ausgegeben werden, je nach eNet-Server events) wieviele Jalousien werden auf einmal bewegt? ( evtl. Zeitproblem???)

**Comment by joergpichler** [comment]
Hier beispielhaft anhand einer Szene die 4 Jalousien schaltet. Max schalte ich 6 Jalousien auf einemal mit einer Szene.

Ausgangssituation:
![grafik](https://user-images.githubusercontent.com/7032094/117301521-013e4280-ae7b-11eb-96c2-6f1e4634a53f.png)
```
[
    {
        "id": "enet.0.Haus.Wohnung.Wohnzimmer.Roll Ost Links #1.Jalousie O L #0.OutputDevice1 #0.value",
        "val": 0,
        "ts": 1620275551992
    },
    {
        "id": "enet.0.Haus.Wohnung.Wohnzimmer.Roll Ost Mitte #1.Jalousie O M #0.OutputDevice1 #0.value",
        "val": 49,
        "ts": 1620275535912
    },
    {
        "id": "enet.0.Haus.Wohnung.Wohnzimmer.Roll Ost Rechts #1.Jalousie O R #0.OutputDevice1 #0.value",
        "val": 0,
        "ts": 1620275556078
    },
    {
        "id": "enet.0.Haus.Wohnung.Wohnzimmer.Roll Süd Links #1.Jalousie S L #0.OutputDevice1 #0.value",
        "val": 0,
        "ts": 1620275549995
    },
    {
        "id": "enet.0.Haus.Wohnung.Wohnzimmer.Roll Süd Rechts #1.Jalousie S R #0.OutputDevice1 #0.value",
        "val": 19,
        "ts": 1620275542812
    }
]
```

Nach schalten der Szene:
![grafik](https://user-images.githubusercontent.com/7032094/117301577-0f8c5e80-ae7b-11eb-8291-cadb9f7e31a0.png)
```
[
    {
        "id": "enet.0.Haus.Wohnung.Wohnzimmer.Roll Ost Links #1.Jalousie O L #0.OutputDevice1 #0.value",
        "val": 0,
        "ts": 1620275551992
    },
    {
        "id": "enet.0.Haus.Wohnung.Wohnzimmer.Roll Ost Mitte #1.Jalousie O M #0.OutputDevice1 #0.value",
        "val": 70,
        "ts": 1620305305654
    },
    {
        "id": "enet.0.Haus.Wohnung.Wohnzimmer.Roll Ost Rechts #1.Jalousie O R #0.OutputDevice1 #0.value",
        "val": 60,
        "ts": 1620305302453
    },
    {
        "id": "enet.0.Haus.Wohnung.Wohnzimmer.Roll Süd Links #1.Jalousie S L #0.OutputDevice1 #0.value",
        "val": 75,
        "ts": 1620305307761
    },
    {
        "id": "enet.0.Haus.Wohnung.Wohnzimmer.Roll Süd Rechts #1.Jalousie S R #0.OutputDevice1 #0.value",
        "val": 75,
        "ts": 1620305309526
    }
]
```

Und hier das Log:
```
2021-05-06 14:48:05.342  - info: enet.0 (11288) Change:Haus.Wohnung.Wohnzimmer.Roll Ost Mitte #1.Jalousie O M #0.OutputDevice1 #0.value  setze auf:1
2021-05-06 14:48:07.734  - info: enet.0 (11288) Change:Haus.Wohnung.Wohnzimmer.Roll Ost Rechts #1.Jalousie O R #0.OutputDevice1 #0.value  setze auf:10
2021-05-06 14:48:09.610  - info: enet.0 (11288) Change:Haus.Wohnung.Wohnzimmer.Roll Süd Links #1.Jalousie S L #0.OutputDevice1 #0.value  setze auf:10
2021-05-06 14:48:11.986  - info: enet.0 (11288) Change:Haus.Wohnung.Wohnzimmer.Roll Ost Mitte #1.Jalousie O M #0.OutputDevice1 #0.value  setze auf:20
2021-05-06 14:48:11.987  - info: enet.0 (11288) Change:Haus.Wohnung.Wohnzimmer.Roll Ost Mitte #1.Jalousie O M #0.OutputDevice1 #0.value  setze auf:20
2021-05-06 14:48:11.988  - info: enet.0 (11288) Change:Haus.Wohnung.Wohnzimmer.Roll Ost Mitte #1.Jalousie O M #0.OutputDevice1 #0.value  setze auf:20
2021-05-06 14:48:13.786  - info: enet.0 (11288) Change:Haus.Wohnung.Wohnzimmer.Roll Ost Mitte #1.Jalousie O M #0.OutputDevice1 #0.value  setze auf:30
2021-05-06 14:48:15.960  - info: enet.0 (11288) Change:Haus.Wohnung.Wohnzimmer.Roll Ost Rechts #1.Jalousie O R #0.OutputDevice1 #0.value  setze auf:40
2021-05-06 14:48:15.962  - info: enet.0 (11288) Change:Haus.Wohnung.Wohnzimmer.Roll Ost Rechts #1.Jalousie O R #0.OutputDevice1 #0.value  setze auf:40
2021-05-06 14:48:17.688  - info: enet.0 (11288) Change:Haus.Wohnung.Wohnzimmer.Roll Süd Links #1.Jalousie S L #0.OutputDevice1 #0.value  setze auf:40
2021-05-06 14:48:20.339  - info: enet.0 (11288) Change:Haus.Wohnung.Wohnzimmer.Roll Ost Mitte #1.Jalousie O M #0.OutputDevice1 #0.value  setze auf:51
2021-05-06 14:48:20.341  - info: enet.0 (11288) Change:Haus.Wohnung.Wohnzimmer.Roll Ost Mitte #1.Jalousie O M #0.OutputDevice1 #0.value  setze auf:51
2021-05-06 14:48:20.341  - info: enet.0 (11288) Change:Haus.Wohnung.Wohnzimmer.Roll Ost Mitte #1.Jalousie O M #0.OutputDevice1 #0.value  setze auf:51
2021-05-06 14:48:22.446  - info: enet.0 (11288) Change:Haus.Wohnung.Wohnzimmer.Roll Ost Rechts #1.Jalousie O R #0.OutputDevice1 #0.value  setze auf:60
2021-05-06 14:48:25.643  - info: enet.0 (11288) Change:Haus.Wohnung.Wohnzimmer.Roll Ost Mitte #1.Jalousie O M #0.OutputDevice1 #0.value  setze auf:70
2021-05-06 14:48:27.751  - info: enet.0 (11288) Change:Haus.Wohnung.Wohnzimmer.Roll Süd Links #1.Jalousie S L #0.OutputDevice1 #0.value  setze auf:75
2021-05-06 14:48:27.753  - info: enet.0 (11288) Change:Haus.Wohnung.Wohnzimmer.Roll Süd Links #1.Jalousie S L #0.OutputDevice1 #0.value  setze auf:75
2021-05-06 14:48:29.517  - info: enet.0 (11288) Change:Haus.Wohnung.Wohnzimmer.Roll Süd Rechts #1.Jalousie S R #0.OutputDevice1 #0.value  setze auf:75
```

**Comment by stoffel7** [comment]
heiß das es müßten alle auf 100% sein?

**Comment by joergpichler** [comment]
Die Screenshots der eNet App zeigen die korrekten Werte: 0, 70, 75, 75, 75
ioBroker zeigt aber: 0, 70, 60, 75, 75

Die Ausgangssituation hat auch nicht gepasst: 0, 0, 0, 0, 0 in der App (korrekte Werte), ioBroker: 0, 49, 0, 0, 19

**Comment by stoffel7** [comment]
D.h. die endgültigen Werte sind weder in den  ioBroker Datenpunkten als auch tauchen sie nicht im log auf...
Ich vermute daß hier der eNetServer etwas „verschluckt“... wird schwierig...

Von meinem iPad gesendet

**Comment by stoffel7** [comment]
nochmals die Frage, tauchen die Änderungen nur in den Datenpunkten nicht auf oder gibt es auch keinen Eintrag im LOG?

**Comment by joergpichler** [comment]
In beidem LOG & Datenpunkte stimmt meist 1-2 Datenpunkte nicht.

Gerade wieder via Szene 4 Jalousien auf 0 gefahren. Jalousie O M müsste auch auf 0 stehen, letzte Aktualisierung ist aber 9.

Log:
enet.0 | 2021-05-12 17:15:53.496 | info | (17058) Change:Haus.Wohnung.Wohnzimmer.Roll Ost Rechts #1.Jalousie O R #0.OutputDevice1 #0.value  setze auf:0
-- | -- | -- | --
enet.0 | 2021-05-12 17:15:50.403 | info | (17058) Change:Haus.Wohnung.Wohnzimmer.Roll Süd Rechts #1.Jalousie S R #0.OutputDevice1 #0.value  setze auf:0
enet.0 | 2021-05-12 17:15:47.945 | info | (17058) Change:Haus.Wohnung.Wohnzimmer.Roll Süd Links #1.Jalousie S L #0.OutputDevice1 #0.value  setze auf:0
enet.0 | 2021-05-12 17:15:46.323 | info | (17058) Change:Haus.Wohnung.Wohnzimmer.Roll Ost Rechts #1.Jalousie O R #0.OutputDevice1 #0.value  setze auf:4
enet.0 | 2021-05-12 17:15:46.314 | info | (17058) Change:Haus.Wohnung.Wohnzimmer.Roll Ost Rechts #1.Jalousie O R #0.OutputDevice1 #0.value  setze auf:4
enet.0 | 2021-05-12 17:15:44.558 | info | (17058) Change:Haus.Wohnung.Wohnzimmer.Roll Ost Mitte #1.Jalousie O M #0.OutputDevice1 #0.value  setze auf:9
enet.0 | 2021-05-12 17:15:42.998 | info | (17058) Change:Haus.Wohnung.Wohnzimmer.Roll Süd Rechts #1.Jalousie S R #0.OutputDevice1 #0.value  setze auf:15
enet.0 | 2021-05-12 17:15:41.351 | info | (17058) Change:Haus.Wohnung.Wohnzimmer.Roll Süd Links #1.Jalousie S L #0.OutputDevice1 #0.value  setze auf:25
enet.0 | 2021-05-12 17:15:39.657 | info | (17058) Change:Haus.Wohnung.Wohnzimmer.Roll Ost Rechts #1.Jalousie O R #0.OutputDevice1 #0.value  setze auf:24
enet.0 | 2021-05-12 17:15:38.007 | info | (17058) Change:Haus.Wohnung.Wohnzimmer.Roll Ost Rechts #1.Jalousie O R #0.OutputDevice1 #0.value  setze auf:35
enet.0 | 2021-05-12 17:15:38.005 | info | (17058) Change:Haus.Wohnung.Wohnzimmer.Roll Ost Rechts #1.Jalousie O R #0.OutputDevice1 #0.value  setze auf:35
enet.0 | 2021-05-12 17:15:36.191 | info | (17058) Change:Haus.Wohnung.Wohnzimmer.Roll Ost Mitte #1.Jalousie O M #0.OutputDevice1 #0.value  setze auf:40
enet.0 | 2021-05-12 17:15:34.487 | info | (17058) Change:Haus.Wohnung.Wohnzimmer.Roll Süd Rechts #1.Jalousie S R #0.OutputDevice1 #0.value  setze auf:45
enet.0 | 2021-05-12 17:15:32.364 | info | (17058) Change:Haus.Wohnung.Wohnzimmer.Roll Süd Links #1.Jalousie S L #0.OutputDevice1 #0.value  setze auf:55
enet.0 | 2021-05-12 17:15:32.362 | info | (17058) Change:Haus.Wohnung.Wohnzimmer.Roll Süd Links #1.Jalousie S L #0.OutputDevice1 #0.value  setze auf:55
enet.0 | 2021-05-12 17:15:30.648 | info | (17058) Change:Haus.Wohnung.Wohnzimmer.Roll Ost Rechts #1.Jalousie O R #0.OutputDevice1 #0.value  setze auf:65
enet.0 | 2021-05-12 17:15:28.700 | info | (17058) Change:Haus.Wohnung.Wohnzimmer.Roll Ost Mitte #1.Jalousie O M #0.OutputDevice1 #0.value  setze auf:60
```

![grafik](https://user-images.githubusercontent.com/7032094/118001180-c5a5eb80-b346-11eb-8361-51e9f2e7ac66.png)

**Comment by stoffel7** [comment]
Das LOG zeigt, daß vom eNet-Server nicht alle Events übermittelt werden, bzw. beim Adapter nicht ankommen, ...
Läuft parallel eine zweite Instanz oder läuft parallel die eNet-App auf einem Smartphone oder Tablet?

**Comment by joergpichler** [comment]
Es läuft nur eine Instanz und die App ist auf dem Smartphone installiert aber wird währenddessen nicht aktiv benutzt. Auch Smartphone währenddessen im Flugmodus ändert am Verhalten leider nichts.

**Comment by stoffel7** [comment]
Ist es wieder die gleiche Situation,
Im LOG fehlen die Werte?

Von meinem iPad gesendet

**Comment by joergpichler** [comment]
Ja. Ich habe den Fall das Log und Datenpunkte voneinander abweichen noch nie beobachtet.

Ist es möglich (als vorübergehenden Workaround) einzubauen, dass nach dem letzten push update vom eNet Server nach kurzer Wartezeit der Wert nochmal gepollt wird?

**Comment by ringel17** [comment]
Hi zusammen, 
ich habe ein ähnliches Verhalten beim Schalten über die Taster (3-fach Taster, über den Server programmiert/verbunden) von Dimmern. Teilweise werden die Werte von AN/AUS und Dimmer-Value inkonsistent gesetzt oder nicht aktualisiert, z.B. Status „AN“ mit Dimmvalue „0“. Dies bleibt konstant „falsch“. Dann kann über andere Adapter (z.B. Yahka) nicht mehr geschaltet werden, da vom Homekit der falsche Status angenommen wird. 

Allerdings: Wenn ich einen Restart des Adapters durchführe, werden nach etwa 30 bis 60 Sekunden die korrekten Werte geholt/gesetzt.

VG, Jörg

**Comment by stoffel7** [comment]
wie schaltest du im iobroker?
über INPUT values oder OUTPUT values?
was hast du als default Wert im eNet Server für Dimmung gesetzt?


Von: joergringelstein ***@***.***>
Gesendet: Sonntag, 30. Mai 2021 22:07
An: SebastianSchultz/ioBroker.enet
Cc: Stefan Bruder; Comment
Betreff: Re: [SebastianSchultz/ioBroker.enet] Sporadisch keine Aktualisierung von Objektdaten (#32)





** CAUTION - email originated from outside the Schaeffler organization






Hi zusammen,
ich habe ein ähnliches Verhalten beim Schalten über die Taster (3-fach Taster, über den Server programmiert/verbunden) von Dimmern. Teilweise werden die Werte von AN/AUS und Dimmer-Value inkonsistent gesetzt oder nicht aktualisiert, z.B. Status „AN“ mit Dimmvalue „0“. Dies bleibt konstant „falsch“. Dann kann über andere Adapter (z.B. Yahka) nicht mehr geschaltet werden, da vom Homekit der falsche Status angenommen wird.

Allerdings: Wenn ich einen Restart des Adapters durchführe, werden nach etwa 30 bis 60 Sekunden die korrekten Werte geholt/gesetzt.

VG, Jörg

**Comment by stoffel7** [comment]
in der ersten Version des  Adapters wurde alle paar Sekunden die Konfiguration neu eingelesen, das geht aber nur bei "kleinen Installationen, bei vielen Devices(Aktoren, etc..) dauert das zu lange... ( man legt u.U. den eNet-Server lahm).

**Comment by ringel17** [comment]
Ich schalte im iobroker über OUTPUT, da ich die beiden Objekte für AN/AUS und Dimmer-Value direkt über YAHKA verbinde. Ich habe auch schon mit Dummy-Objekten für die YAHKA-Seite gearbeitet und mit Skripten eine "vermutete" Synchronisation programmiert.

Das Schalten über den iobroker funktioniert (meistens) gut, solange aktiv die OUTPUT-Werte über den iobroker geschaltet werden. Wenn Taster betätigt oder Szenen aktiviert werden, wird der Dimmer-Value oft aktualisiert. Der ON/OFF-Wert sehr selten konsistent nachgezogen. Es gibt aber auch Situationen, bei denen der ON/OFF-Wert ad hoc geschaltet wird und der Dimmer-Value auf einem falschen Wert verharrt. 

Die Log-Einträge passen zum Objektstatus.

Meinst du als Default-Wert die EInschalthelligeit? Sie ist auf 100% gesetzt.

Aus meiner Sicht liegt das Problem beim eNet-Server, welcher die Werte nicht konsistent kommuniziert. Deshalb hatte ich leichtsinnig ein erneutes Einlesen der Konfiguration vorgeschlagen.

**Comment by stoffel7** [comment]
Ja, das scheint aktuell wirklich so zu sein, der eNet-Server kommuniziert nicht sauber auf die "changes".
Es scheint auch von der Firmware auf dem Server abhängig zu sein, ich hab noch die V2.0, da funktioniert das bis auf eine Ausnahme(ist ein Dimmer mit Nebenstelleneingang) problemlos.
Es gibt dafür eigentlich die functionen "registerEvent..."", die sollen genau die Änderungen an Device values zurück liefern.
Leider bekommen wir keine Unterstützung dazu vom Hersteller.
